packages/data/src/onboarding/resolvers.js

```diff
@@ -60,14 +68,77 @@ export function* getTasksStatus() {
 	}
 }
 
+function getQuery() {
+	const searchString = window.location && window.location.search;
+	if ( ! searchString ) {
+		return {};
+	}
+
+	const search = searchString.substring( 1 );
+	return parse( search );
+}
+
+/**
+ * This function will be depreciated in favour of registering tasks on the back-end.
+ *
+ * @param {Array} taskLists array of task lists.
+ */
+function* mergeWithFilteredTasks( taskLists ) {
+	const filteredTasks = applyFilters(
+		'woocommerce_admin_onboarding_task_list',
+		[],
+		getQuery()
+	);
+	if ( filteredTasks && filteredTasks.length > 0 ) {
+		deprecated( 'woocommerce_admin_onboarding_task_list', {
+			version: '2.10.0',
+			alternative: 'TaskLists::add_task()',
+			plugin: '@woocommerce/data',
+		} );
+		// Show depreciation notice;
+		const dismissedTasks = yield resolveSelect(
```

Comment:
These resolvers are kind of a bummer.  I know they'll eventually get removed, but out of curiosity are any of the known extended tasks using dismissal or snoozing?

We might also look into be able to resolve async given these two `resolveSelect` calls have to completely resolve synchronously, right?  Might add a decent amount of load time.

Reply:
> These resolvers are kind of a bummer. I know they'll eventually get removed, but out of curiosity are any of the known extended tasks using dismissal or snoozing?

Yeah, most of them support dismissing and wc pay has a couple that can be snoozed as well.

> We might also look into be able to resolve async given these two resolveSelect calls have to completely resolve synchronously, right? Might add a decent amount of load time.

Yeah it would need to be done synchronously, otherwise it causes weird behaviour. One thing that I was thinking otherwise is passing the `ids` (or the entire object) to the `/onboarding/tasks` request, and create those on the fly. This automatically make set the snooze/dismiss related variables. @joshuatf thoughts?

Reply:
@louwie17 I really like the idea of creating these tasks on the fly.  This solves a lot of our problems and keeps things relatively clean and easy to remove going forward.